docs/user/gcp/install_upi.md

```diff
@@ -0,0 +1,589 @@
+
+resources:
+- name: cluster-control-plane
+  type: 05_control_plane.py
+  properties:
+    infra_id: '${INFRA_ID}'
+    region: '${REGION}'
+
+    control_subnet: '${CONTROL_SUBNET}'
+    image: '${CLUSTER_IMAGE}'
+    machine_type: 'n1-standard-4'
+    root_volume_size: '128'
+    service_account_email: '${MASTER_SERVICE_ACCOUNT_EMAIL}'
+
+    ignition: '${MASTER_IGNITION}'
+EOF
+```
+- infra_id: the infra_id from above (ie openshift-xxxx)
+- region: the region to deploy the cluster into (ie us-east1)
+- control_subnet: the api path to the control subnet
+- image: the api path to the RHCOS image
+- machine_type: The machine type of the instance (ie n1-standard-4)
+- service_account_email: the email address for the master service account created above
+- ignition: the contents of the master.ign file
+
+Create the deployment using gcloud.
+
+```sh
+gcloud deployment-manager deployments create ${INFRA_ID}-control-plane --config 05_control_plane.yaml
+```
+
+The templates do not manage DNS entries due to limitations of Deployment
+Manager, so we must add the etcd entries manually.
+
+```sh
+export MASTER0_IP=`gcloud compute instances describe ${INFRA_ID}-m-0 --zone ${REGION}-a --format json | jq -r .networkInterfaces[0].networkIP`
+export MASTER1_IP=`gcloud compute instances describe ${INFRA_ID}-m-1 --zone ${REGION}-b --format json | jq -r .networkInterfaces[0].networkIP`
+export MASTER2_IP=`gcloud compute instances describe ${INFRA_ID}-m-2 --zone ${REGION}-c --format json | jq -r .networkInterfaces[0].networkIP`
+if [ -f transaction.yaml ]; then rm transaction.yaml; fi
+gcloud dns record-sets transaction start --zone ${INFRA_ID}-private-zone
+gcloud dns record-sets transaction add "${MASTER0_IP}" --name etcd-0.${CLUSTER_NAME}.${BASE_DOMAIN}. --ttl 60 --type A --zone ${INFRA_ID}-private-zone
+gcloud dns record-sets transaction add "${MASTER1_IP}" --name etcd-1.${CLUSTER_NAME}.${BASE_DOMAIN}. --ttl 60 --type A --zone ${INFRA_ID}-private-zone
+gcloud dns record-sets transaction add "${MASTER2_IP}" --name etcd-2.${CLUSTER_NAME}.${BASE_DOMAIN}. --ttl 60 --type A --zone ${INFRA_ID}-private-zone
+gcloud dns record-sets transaction add \
+  "0 10 2380 etcd-0.${CLUSTER_NAME}.${BASE_DOMAIN}." \
+  "0 10 2380 etcd-1.${CLUSTER_NAME}.${BASE_DOMAIN}." \
+  "0 10 2380 etcd-2.${CLUSTER_NAME}.${BASE_DOMAIN}." \
+  --name _etcd-server-ssl._tcp.${CLUSTER_NAME}.${BASE_DOMAIN}. --ttl 60 --type SRV --zone ${INFRA_ID}-private-zone
+gcloud dns record-sets transaction execute --zone ${INFRA_ID}-private-zone
+```
+
+## Monitor for `bootstrap-complete`
+
+```console
+$ openshift-install wait-for bootstrap-complete
+INFO Waiting up to 30m0s for the Kubernetes API at https://api.test.example.com:6443...
+INFO API v1.12.4+c53f462 up
+INFO Waiting up to 30m0s for the bootstrap-complete event...
+```
+## Pivot load balancers to control plane
+
+The bootstrap node can now be removed from the load balancers. First, the
+control plane needs to be added.
```

Comment:
The following block both adds control-plane targets and removes the bootstrap target.  Either say that you're about to do both, or split it up like:

    First, the control plane needs to be added:

    ```sh
    ...
    ```

    Then the bootstrap targets can be removed:

    ```sh
    ...
    ```

Reply:
Done.